pkg/controller/jobframework/setup.go

```diff
@@ -67,24 +91,14 @@ func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
 			if err != nil {
 				return fmt.Errorf("%s: %w: %w", fwkNamePrefix, errFailedMappingResource, err)
 			}
-			if _, err = mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
-				if !meta.IsNoMatchError(err) {
-					return fmt.Errorf("%s: %w", fwkNamePrefix, err)
-				}
-				logger.Info("No matching API in the server for job framework, skipped setup of controller and webhook")
+			if !isAPIAvailable(mgr, RayCRDsName(), TrainingOperatorCRDsName()) {
+				logger.Info("API not available, waiting for it to become available... - Skipping setup of controller and webhook")
+				waitForAPIs(context.TODO(), logger, mgr, gvk, RayCRDsName(), TrainingOperatorCRDsName(), func() {
```

Comment:
Nit:
```suggestion
				waitForAPIs(context.Background(), logger, mgr, gvk, RayCRDsName(), TrainingOperatorCRDsName(), func() {
```

TODO returns a non-nil and an empty Context generally used as a placeholder. Ideally it's good to create a single cancellable context for the entire reconciler so that if one gets cancelled, it's easier to stop all the other processes. But given we are doing this in the manager, I'd suggest to use `context.Background()` - which initialises a new one, that at least doesn't get cancelled till the manager stops (though not perfectly ideal, but safer than leaving it as a placeholder given that we know where it is being used).